### ISSUE TITLE
Add new icon in header of table

### ISSUE THREAD

**Issue by vantsymbalenko** [gregnb/mui-datatables#129]
How can I add new icon with custom handler in header of table near search icon? Is it possible?

**Comment by NickToye** [comment]
customToolbar is your friend.

**Comment by SoftwireTimC** [comment]
When I add a custom toolbar it adds my new toolbar but it puts it inside the old one instead of remove the previous one. Is the idea of custom toolbar to make a completely new toolbar or is it just for adding extra buttons to the standard toolbar?

**Comment by NickToye** [comment]
I believe it’s to add more.

**Comment by gabrielliwerant** [comment]
@vantsymbalenko Your example uses redux, so you need to set up redux properly in your app. This is not a problem with the `customToolbar` functionality, as it works as intended in the example in the application. If there is a bug with the `customToolbar`, you can open up a new issue with steps to reproduce it and/or a minimal example.

**Comment by DigiGroundJohn** [comment]
Hi,

If I pass the function as property if didn't work for me 

   onViewStoriesClick(){
       console.log("button clikc is working")
      this.setStoryDialog(true);
   }


<CustomToolbar text={"View New Stories"} onClick={()=>this.onViewStoriesClick() } cssClasses={"blue-button"}/>


and my custom toolbar is 

render() {
    // const { classes } = this.props;
console.log(this.props)
    return (
      <React.Fragment>
        <Tooltip title={this.props.text}>
               
                <Button onClick={()=>this.props.onClick} className={this.props.cssClasses}>
                    {/* <AddIcon className={classes.deleteIcon} /> */}
                    {/* <SvgIcon> */}
                      {this.props.text}
                    {/* </SvgIcon> */}
                </Button>
        </Tooltip>
      </React.Fragment>
    );
  }

}

**Comment by gabrielliwerant** [comment]
You've made things a little more complicated than they need to be. You're passing an anonymous function that runs your click handler, instead of just the function. It's also a bit confusing to assign the prop to the click handler name.

Try something like 
```
myClickHandler = () => console.log('clicked!');

<MyComponent clickHandler={this.myClickHandler />

...

<IconButton onClick={() => this.props.myClickHandler()}>
```

Or if you wanted to avoid creating a new anonymous function in your toolbar component, you can use something like
```
myClickHandler = () => () => console.log('clicked!');

<MyComponent clickHandler={this.myClickHandler } />

...

<IconButton onClick={this.props.myClickHandler()}>
```

**Comment by DigiGroundJohn** [comment]
@gabrielliwerant 
Ahh thanks, to point out this think.

Actually I am doing different action according to parent component that why it becomes complex, but it is working now. 

awesome. Thanks

**Comment by usahai** [comment]
Hello! I've used your tables across 2 projects and found it fantastic! Thanks a lot for building this and the constant support.

I had a question about the custom toolbar - specifically, the tooltip for a new custom action. It does not seem to inherit the properties that the parent component (the table) has. As such, I'm getting a massive tooltip for the new component. Is there anyway it can inherit the CSS of the original?

`		return (
			<React.Fragment>
                <ThemeProvider theme={theme}>
                    <Tooltip title={"Download"}>
                        <IconButton className={classes.iconButton} onClick={this.handleClick}>
                            <AddIcon />
                        </IconButton>
                    </Tooltip>
                </ThemeProvider>
			</React.Fragment>
		);
`
I have also tried it by removing the theme provider just in case there was an overriding property, but it didn't work. Any suggestions?
